pkg/build/reaper/reaper_test.go

```diff
@@ -0,0 +1,195 @@
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+				ktestclient.NewDeleteAction("builds", "default", "build-1"),
+				ktestclient.NewDeleteAction("builds", "default", "build-3"),
+				ktestclient.NewDeleteAction("builds", "default", "build-2"),
+				ktestclient.NewDeleteAction("builds", "default", "build-4"),
+			},
+			output: "config stopped",
+			err:    false,
+		},
+		{
+			testName:  "no config, some builds",
+			namespace: "default",
+			name:      "config",
+			oc:        newBuildListFake(mkbuildlist(2)),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+				ktestclient.NewDeleteAction("builds", "default", "build-1"),
+				ktestclient.NewDeleteAction("builds", "default", "build-2"),
+			},
+			output: "config stopped",
+			err:    false,
+		},
+		{
+			testName:  "no config, no builds",
+			namespace: "default",
+			name:      "config",
+			oc:        testclient.NewSimpleFake(notfound()),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+			},
+			output: "",
+			err:    true,
+		},
+		{
+			testName:  "config, no builds",
+			namespace: "default",
+			name:      "config",
+			oc:        testclient.NewSimpleFake(mkbuildconfig(5)),
+			expected: []ktestclient.Action{
+				ktestclient.NewDeleteAction("buildconfigs", "default", "config"),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelector("config"), nil),
+				ktestclient.NewListAction("builds", "default", buildutil.ConfigSelectorDeprecated("config"), nil),
+			},
+			output: "config stopped",
+			err:    false,
+		},
+	}
+
+	for _, test := range tests {
+		reaper := &BuildConfigReaper{oc: test.oc, pollInterval: time.Millisecond, timeout: time.Millisecond}
+		out, err := reaper.Stop(test.namespace, test.name, 1*time.Second, nil)
+
+		if !test.err && err != nil {
+			t.Errorf("%s: unexpected error: %v", test.testName, err)
+		}
+		if test.err && err == nil {
+			t.Errorf("%s: expected an error", test.testName)
+		}
+		if len(test.oc.Actions()) != len(test.expected) {
+			t.Fatalf("%s: unexpected actions: %v, expected %v", test.testName, test.oc.Actions(), test.expected)
+		}
+		for j, actualAction := range test.oc.Actions() {
+			if !reflect.DeepEqual(actualAction, test.expected[j]) {
+				t.Errorf("%s: unexpected action: %s, expected %s", test.testName, actualAction, test.expected[j])
+			}
+		}
+		if out != test.output {
```

Comment:
Drop this test, we've changed the Reaper interface upstream already, currently it returns only error. This will cause me problems upon next rebase ;)